lib/child_process.js

```diff
@@ -161,15 +168,40 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
+  return (file, options = {}) => {
     let resolve;
     let reject;
     const promise = new Promise((res, rej) => {
       resolve = res;
       reject = rej;
     });
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
+    if (typeof options !== 'object') {
+      return Promise.reject(
+        new ERR_INVALID_ARG_TYPE(
+          'options',
+          'Object',
+          options));
+    }
+
+    const { signal } = options;
+    const hasSignal = signal != null;
+    const isNotValidSignal = typeof signal !== 'object' ||
+      !('aborted' in signal);
+
+    if (hasSignal) {
+      if (isNotValidSignal) return Promise.reject(
+        new ERR_INVALID_ARG_TYPE(
+          'options.signal',
+          'AbortSignal',
+          signal));
+
+      signal.addEventListener('abort', () => {
+        reject(lazyDOMException('AbortError'));
+      }, { once: true });
```

Comment:
The event listener should be cleared once the Promise is resolved to ensure that triggering the `AbortSignal` after the Promise is resolved does not cause a rejection after.

Reply:
Also, make sure that the event listener is cleared once the child process terminates, even if the signal has not been triggered.